Multi-line comments inside <p> tags are treated as a single sentence

### Summary

Multi-line comments within `<p>` tags are processed as a single continuous sentence. This causes `sentence-length` rules to trigger errors even when the comment consists of multiple short lines.

### Example

```html
<!--
Comment outside of p tag.
This is ignored.
Even when this is too long and exceeds the maximum sentence length.
long long long long long long long long long long long long long long long long long long long long long long.
-->

<p>
<!--
Comment inside of p tag.
This is processed as a single sentence.
Even when the comment consists of multiple short lines.
Error: Line 9 sentence length(136) exceeds the maximum sentence length of 100.
-->
</p>
```

```sh
$ pnpm textlint test.html
/path/to/test.html
  7:1  error  Line 7 sentence length(202) exceeds the maximum sentence length of 100.
Over 102 characters  japanese/sentence-length

✖ 1 problem (1 error, 0 warnings, 0 infos)
```

### Inconsistent Behavior

Currently, the handling of comments varies depending on the language and context:

- **Markdown**: Comments are ignored by `sentence-length` and do not trigger errors.
- **HTML**:
  - **Inside `<p>` tags**: The entire comment is treated as one sentence.
  - **Outside `<p>` tags**: Ignored.

This inconsistency seems to stem from the following:

- **In HTML**:
  1. Comments are represented as `Comment` nodes.
  2. `sentence-splitter` treats `Comment` nodes as a single sentence.
  3. Therefore, comments inside `<p>` are evaluated as sentences, while those outside `<p>` are ignored because they are not within a Paragraph node.
- **In Markdown**:
  1. Comments are represented as `HTML` nodes.
  2. Since these nodes are typically not inside a Paragraph, they are ignored.

--- a/src/sentence-length.ts
+++ b/src/sentence-length.ts
@@ -119,7 +119,16 @@ const reporter: TextlintRuleReporter<Options> = (context, options = {}) => {
                           })
                       }
                     : sentence;
-                const source = new StringSource(filteredSentence);
+                const source = new StringSource(filteredSentence, {
+                    // https://github.com/textlint-rule/textlint-rule-sentence-length/issues/45
+                    // Treat Comment nodes as zero-length strings
+                    replacer({ node, emptyValue }) {
+                        if (node.type === "Comment") {
+                            return emptyValue();
+                        }
+                        return;
+                    }
+                });
                 const actualText = source.toString();
                 const sentenceText = removeRangeFromString(actualText, skipPatterns);
                 // larger than > 100

--- a/test/sentence-length-html-test.ts
+++ b/test/sentence-length-html-test.ts
@@ -33,6 +33,23 @@ const tester = new TextLintTester();
                 {
                     text: "<p>TEST is <a href='https://example.com'>https://example.com</a></p>",
                     ext: ".html"
+                },
+                // https://github.com/textlint-rule/textlint-rule-sentence-length/issues/45
+                // Comment inside <p> should be ignored (length 0)
+                {
+                    text: "<p><!-- this is a very long comment that should be ignored --></p>",
+                    ext: ".html"
+                },
+                {
+                    text: `<p>
+<!--
+comment-->
+</p>`,
+                    ext: ".html"
+                },
+                {
+                    text: "<p>short text<!-- long long long long long long comment --></p>",
+                    ext: ".html"
                 }
             ],
             invalid: [

